firestore.rules

```diff
@@ -419,15 +419,17 @@ service cloud.firestore {
         function userOwnsDocument() {
           return getDocumentOwner() == string(request.auth.token.platform_user_id);
         }
+
+        function hasDocumentAccess() {
+          return teacherCanAccessDocument() || researcherCanAccessDocument() || userCanAccessDocument();
+        }
+
         match /comments/{commentId} {
-          // portal-authenticated teachers with access to the document can create valid comments
-          allow create: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && isValidCommentCreateRequest();
-          // teachers can only update their own comments and only if they're valid
-          allow update: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && isValidCommentUpdateRequest();
-          // teachers can only delete their own comments
-          allow delete: if (teacherCanAccessDocument() || researcherCanAccessDocument()) && userIsResourceUser();
-          // only teachers that "own" the document can read the comments (for now)
-          allow read: if teacherCanAccessDocument() || researcherCanAccessDocument();
+          // Any user with access to the document can read or create comments and make valid update/delete requests
+          allow create: if hasDocumentAccess() && isValidCommentCreateRequest();
```

Comment:
I'm not sure this is a real problem, but both `teacherCanAccessDocument()` and `researcherCanAccessDocument()` allow comment creation even if the document doesn't exist. But `userCanAccessDocument()` doesn't work in that case.

Perhaps that isn't a problem anymore because we are creating these "metadata" documents in firestore before we try to add comments to them? 

Because of the multiple ways these metadata documents could be named it seems likely that there will be cases where the metadata document (hopefully simple name that is just the CLUE document id), doesn't exist yet.

But perhaps students just don't have access to documents like this so they can't comment on them.